File: src/types/prompt.ts
// prompt.ts — Prompt-related TypeScript types
// Keep in sync with Rust models in src-tauri/src/models/

export type Prompt = {
  id: number;
  title: string;
  content: string;
  description?: string | null;
  groupId?: number | null;
  groupName?: string | null;
  tags: string[];
  isFavorite: boolean;
  usageCount: number;
  lastUsedAt?: string | null;
  createdAt: string;
  updatedAt: string;
};

export type CreatePromptInput = {
  title: string;
  content: string;
  description?: string | null;
  groupId?: number | null;
  tags?: string[];
  isFavorite?: boolean;
};

export type UpdatePromptInput = {
  id: number;
  title?: string;
  content?: string;
  description?: string | null;
  groupId?: number | null;
  tags?: string[];
  isFavorite?: boolean;
};

export type SearchPromptInput = {
  keyword: string;
  groupId?: number | null;
  onlyFavorite?: boolean;
  limit?: number;
};

export type Group = {
  id: number;
  name: string;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
};

export type Tag = {
  id: number;
  name: string;
  createdAt: string;
};

export type UsageLog = {
  id: number;
  promptId: number;
  action: string;
  createdAt: string;
};

File: src-tauri/src/models/prompt.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Prompt {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub description: Option<String>,
    pub group_id: Option<i64>,
    pub is_favorite: bool,
    pub usage_count: i64,
    pub last_used_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

/// DTO returned to frontend — includes joined group name and tags.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PromptDto {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub description: Option<String>,
    pub group_id: Option<i64>,
    pub group_name: Option<String>,
    pub tags: Vec<String>,
    pub is_favorite: bool,
    pub usage_count: i64,
    pub last_used_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreatePromptInput {
    pub title: String,
    pub content: String,
    pub description: Option<String>,
    pub group_id: Option<i64>,
    pub tags: Option<Vec<String>>,
    pub is_favorite: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdatePromptInput {
    pub id: i64,
    pub title: Option<String>,
    pub content: Option<String>,
    pub description: Option<String>,
    pub group_id: Option<i64>,
    pub tags: Option<Vec<String>>,
    pub is_favorite: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SearchPromptInput {
    pub keyword: String,
    pub group_id: Option<i64>,
    pub only_favorite: Option<bool>,
    pub limit: Option<i64>,
}
